lib/optimizely.rb

```diff
@@ -70,7 +70,7 @@ def activate(experiment_key, user_id, attributes = nil)
       @logger.log(Logger::INFO,
                   'Dispatching impression event to URL %s with params %s.' % [impression_event.url,
                                                                               impression_event.params])
-      @event_dispatcher.dispatch_event(impression_event.url, impression_event.params)
+      @event_dispatcher.dispatch_event(impression_event.method, impression_event.url, impression_event.params)
```

Comment:
this is more of a design thing, but semantically speaking, when you say dispatch_event, it makes more sense to pass in the event object instead of splitting it up. As we evolve the SDK I can see this method signature becoming too long/unbearable. Might as well make the refactor here.